File: libnetwork/src/main/java/com/gystry/libnetwork/GetRequest.java
package com.gystry.libnetwork;

/**
 * @author gystry
 * 创建日期：2020/7/3 17
 * 邮箱：gystry@163.com
 * 描述：
 */
public class GetRequest<T> extends Request<T, GetRequest> {
    public GetRequest(String url) {
        super(url);
    }

    @Override
    protected okhttp3.Request generateRequest(okhttp3.Request.Builder builder) {
        //将params拼接到url中
        String urlFromParams = UrlCreater.createUrlFromParams(mUrl, params);
        return null;
    }
}

File: libnetworkkt/src/main/java/com/gystry/libnetworkkt/GetRequest.kt
package com.gystry.libnetworkkt

/**
 * @author gystry
 * 创建日期：2021/3/18 15
 * 邮箱：gystry@163.com
 * 描述：
 */
class GetRequest<T>(url:String):Request<T,GetRequest<T>>(url) {
    override fun generateRequest(builder: okhttp3.Request.Builder): okhttp3.Request {
        val urlFormParams = createUrlFromParams(mUrl, params)
        return builder.url(urlFormParams).get().build()
    }
}